index.js

```diff
@@ -605,6 +605,11 @@
            other.unicode === this.unicode;
   }
 
+  //  RegExp$prototype$lte :: RegExp ~> RegExp -> Boolean
+  function RegExp$prototype$lte(other) {
+    return lte(toString(this), toString(other));
+  }
```

Comment:
Do you think a string comparison is reasonable, or are you unhappy with the way it smushes source and flags together?

Another option would be to sort `/abc/` as `['', 'abc']` and `/def/g` as `['g', 'def']`.